controllers/reconcile.go

```diff
@@ -919,6 +919,56 @@ func setupServiceTilesData(feList *crd.FrontendList, feEnvironment crd.FrontendE
 	return categories, skippedTiles
 }
 
+func getNavItemPath(feName string, bundleID string, segmentID string) string {
+	return fmt.Sprintf("%s-%s-%s", feName, bundleID, segmentID)
+}
+
+func setupBundlesData(feList *crd.FrontendList, feEnvironment crd.FrontendEnvironment) ([]crd.FrontendBundlesGenerated, []string) {
+	bundles := []crd.FrontendBundlesGenerated{}
+	if feEnvironment.Spec.Bundles == nil {
+		// skip if we do not have bundles in fe environment
+		return bundles, []string{}
+	}
+
+	skippedNavItemsMap := make(map[string][]string)
+	bundleNavSegmentMap := make(map[string][]crd.NavigationSegment)
+	for _, frontend := range feList.Items {
+		if frontend.Spec.FeoConfigEnabled && frontend.Spec.NavigationSegments != nil {
+			for _, navSegment := range frontend.Spec.NavigationSegments {
+				bundleNavSegmentMap[navSegment.BundleID] = append(bundleNavSegmentMap[navSegment.BundleID], *navSegment)
+				skippedNavItemsMap[navSegment.BundleID] = append(skippedNavItemsMap[navSegment.BundleID], getNavItemPath(frontend.Name, navSegment.BundleID, navSegment.SegmentID))
+			}
+		}
+	}
+
+	for _, bundle := range *feEnvironment.Spec.Bundles {
+		delete(skippedNavItemsMap, bundle.ID)
+		sort.Slice(bundleNavSegmentMap[bundle.ID], func(i, j int) bool {
+			if (bundleNavSegmentMap[bundle.ID])[i].Position == (bundleNavSegmentMap[bundle.ID])[j].Position {
+				return (bundleNavSegmentMap[bundle.ID])[i].SegmentID[0] < (bundleNavSegmentMap[bundle.ID])[j].SegmentID[0]
```

Comment:
@Hyperkid123 in hindsight - I'm not sure if this is what you meant by falling back to "alphabetical" ordering in case of position number conflict (I'm sorting alphabetically on the `SegmentID`). Could you take a look here? I did update the kuttl test and it is working.

Reply:
If needed we can revert my recent commit for the alphabetical ordering and deliver

Reply:
Yeah, I think this is good enough, I see this just as a fallback to make sure the nav items are "somewhat" in the right place. We won't have many of them as the services will probably very quickly notice the items are not in the right place.